react-search-component/my-app/src/List.tsx

```diff
@@ -0,0 +1,51 @@
+import './List.css';
+
+type StringList = {
+  list: string[];
+  word: string;
+};
+
+export function List({ list, word }: StringList) {
+  const newList = [];
+  let check;
+  for (let i = 0; i < list.length; i++) {
+    if (list[i].toLowerCase().includes(word.toLowerCase())) {
+      check = true;
+      break;
+    } else {
+      check = false;
+    }
+  }
+  for (let i = 0; i < list.length; i++) {
+    newList.push(
+      <li
+        className={
+          !word
+            ? ''
+            : list[i].toLowerCase().includes(word.toLowerCase())
+            ? ''
+            : 'hidden'
+        }
+        key={i}
+        style={{
+          textAlign: 'left',
+        }}>
+        {list[i]}
+      </li>
+    );
+  }
```

Comment:
Not bad. You worked out all the logic to get it to display properly.

The easiest way to do this, though, is to use Array.filter. Something like `newList = list.filter((item) => matches)` where you replace `matches` with the logic to see if the item matches the word.